bsp/stm32/stm32f411-st-nucleo/applications/arduino_pinout/pins_arduino.c

```diff
@@ -46,3 +46,29 @@ const pin_map_t pin_map_table[]=
     {A6, RT_NULL, "adc1", RT_ADC_INTERN_CH_VREF},   /* ADC, On-Chip: internal reference voltage */
     {A7, RT_NULL, "adc1", RT_ADC_INTERN_CH_TEMPER}, /* ADC, On-Chip: internal temperature sensor */
 };
+
+void pins_switch_to_spi(const char *bus_name)
+{
+    if(!rt_strcmp(bus_name, RTDUINO_DEFAULT_SPI_BUS_NAME))
+    {
+        __HAL_RCC_TIM1_CLK_DISABLE();
+        HAL_GPIO_DeInit(GPIOA, GPIO_PIN_7);
+
+        GPIO_InitTypeDef GPIO_InitStruct = {0};
+        __HAL_RCC_SPI1_CLK_ENABLE();
+        __HAL_RCC_GPIOA_CLK_ENABLE();
+        /**SPI1 GPIO Configuration
+        PA5     ------> SPI1_SCK
+        PA6     ------> SPI1_MISO
+        PA7     ------> SPI1_MOSI
+        */
+        GPIO_InitStruct.Pin = GPIO_PIN_5 | GPIO_PIN_6 | GPIO_PIN_7;
+        GPIO_InitStruct.Mode = GPIO_MODE_AF_PP;
+        GPIO_InitStruct.Pull = GPIO_NOPULL;
+        GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
+        GPIO_InitStruct.Alternate = GPIO_AF5_SPI1;
+        HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+        rt_kprintf("D11, D12 and D13 will switch from PWM to SPI\n");
```

Comment:
rt_kprintf仅用于终端交互场景，信息提示用LOG_x
```suggestion
        LOG_I("D11, D12 and D13 will switch from PWM to SPI\n");
```

文件上面别忘了加tag
```
#define DBG_TAG    "RTduino.pins_arduino"
#define DBG_LVL    DBG_INFO
#include <rtdbg.h>
```